pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -452,15 +452,9 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Name: "Sync public service account signing keys to Garden cluster",
 			Fn:   botanist.SyncPublicServiceAccountKeys,
 			SkipIf: o.Shoot.HibernationEnabled ||
-				!features.DefaultFeatureGate.Enabled(features.ShootManagedIssuer) ||
 				!v1beta1helper.HasManagedIssuer(botanist.Shoot.GetInfo()),
 			Dependencies: flow.NewTaskIDs(initializeShootClients),
 		})
-		_ = g.Add(flow.Task{
-			Name:   "Delete public service account signing keys from Garden cluster",
-			Fn:     botanist.DeletePublicServiceAccountKeys,
-			SkipIf: features.DefaultFeatureGate.Enabled(features.ShootManagedIssuer),
```

Comment:
Why is this task removed, should it be just un-skipped?

Reply:
Previously we were deleting all already synched keys if the feature gate was disabled for some reason. With this proposed change we no longer have the feature gate and the keys will only be synched to the Garden cluster if one enables the feature for a shoot cluster. Once enabled the feature cannot be disabled. This means that we only have to worry about removing the synched keys during shoot deletion. An alternative would be to have a deletion call if the feature is not enabled for a shoot cluster, but that would result in a noop most of the times.